x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,209 @@
+             - extract_bro_specifics
+
+
+  #################################################################################################
+  # System EPM package
+  - id: system
+    title: Collect system information and metrics
+    package:
+      name: epm/system
+      version: 1.7.0
+    inputs:
+      - type: metrics/system
+        streams:
+         - id?: {id}
+           enabled?: false # default true
+           metricset: cpu
+           period: 10s
+           dataset: system.cpu
+           metrics: ["percentages", "normalized_percentages"]
+         - metricset: memory
+           dataset: system.memory
+         - metricset: diskio
+           dataset: system.diskio
+         - metricset: load
+           dataset: system.load
+         - metricset: memory
+           dataset: system.memory
+         - metricset: process
+           dataset: system.process
+           processes: ["firefox*"]
+         - metricset: process_summary
+           dataset: system.process_summary
+         - metricset: uptime
+           dataset: system.uptime
+         - metricset: socket_summary
+           dataset: system.socket_summary
+         - metricset: filesystem
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+#################################################################################################
+### suggestion 1
+ - id: myendpoint-x1
+   title: Endpoint configuration
+   package:
+     name: endpoint
+     version: xxx
+   inputs:
+     - type: endpoint # Reserved key word
+       streams:
+         - type: malware
+           detect: true
+           prevent: false
+           notify_user: false
+           threshold: recommended
+           platform: windows
+
+        - type: eventing
+          api: true
+          clr: false
+          dll_and_driver_load: false
+          dns: true
+          file: false
+          platform: windows
+
+       - type: malware
+         detect: true
+         prevent: false
+         notify_user: false
+         threshold: recommended
+         platform: mac
+
+      - type: eventing
+        api: true
+        clr: false
+        dll_and_driver_load: false
+        dns: true
+        file: false
+        platform: mac
+
+       - type: malware
+         detect: true
+         prevent: false
+         notify_user: false
+         threshold: recommended
+         platform: linux
+
+      - type: eventing
+        api: true
+        clr: false
+        dll_and_driver_load: false
+        dns: true
+        file: false
+        platform: linux
+
+#################################################################################################
+### suggestion 2
+ - id: myendpoint-1
```

Comment:
do the individual `id`s in here imply that we'll edit individual sections on the Agent configuration?  Is there an overall Agent configuration ID that we'll use with Ingest API's edit the entire configuration?

I'm concerned about editing the configuration in pieces, it's preferable to send the entire configuration down to the Agent/Endpoint at once.

Reply:
There will be an overall unique ID for a configuration in the Fleet API. These "local" ids can be generated and become useful especially on error reporting, so we can tell the user which part of the config did not work.

It is still the plan to send down the full agent config down to the agent at once.